File: microservices/Models/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EventManagementSystem.Models
{
    [Table("Events", Schema = "EventManagement")]
    public class Event
    {
        [Key]
        public int event_id { get; set; }

        [Required]
        [StringLength(300)]
        public string name { get; set; } = string.Empty;

        public string? description { get; set; }

        public DateTime start_date { get; set; }
        public DateTime? end_date { get; set; }

        [StringLength(300)]
        public string? venue { get; set; }

        public int? capacity { get; set; }

        [Required]
        [StringLength(20)]
        public string status { get; set; } = "Active"; // Changed to string

        public int created_by { get; set; }
    }
}

File: front-end/src/models/event.ts
// src/models/event.ts

export interface EventResponseDto {
  event_id: number;
  name: string;
  description: string;
  start_date: string; // ISO 8601 format: "2025-01-10T00:00:00"
  end_date: string;   // ISO 8601 format: "2025-01-12T00:00:00"
  venue: string;
  capacity: number;
  status: string;     // e.g., "Published", "Draft", "Approved", "Rejected"
  created_by: number; // User ID of event creator
}

// Frontend interface for displaying events
export interface EventForDisplay extends EventResponseDto {
  id: string; // Convert event_id to string
  formattedStartDate?: string;
  formattedEndDate?: string;
  organizerName?: string;
  organizerEmail?: string;
  remainingSeats?: number;
  totalSeats?: number;
  image?: string;
  schedule?: { day: number; sessions: any[] }[];
}